### ISSUE TITLE
Ability to use different indent logic

### ISSUE THREAD

**Issue by sten22** [Camelcade/Perl5-IDEA#1406]
This is how Camelcade format indent by default, here we have extra indent for sub block:
https://monosnap.com/file/HfhdU0O295S03X7WSam42tvUJZlbQX

This is uncommon style, usualy we keep braces on same level and use one level indent on sub body: https://github.com/kraih/mojo/blob/master/lib/Mojo.pm

Nested subs can cause "indent hell":
https://monosnap.com/file/6HY4QrUlmDoOgldRgNOx8j74NSTQBV

Could you add ability to confiruge editor like this:
https://monosnap.com/file/i99yqWTrQQJNKTRTrXStook7nVAPA4

- keep close braces, brackets, square brackets on same level
- indent sub body, args list, list definitions, etc. only once

**Comment by hurricup** [comment]
The question is how to be with sequences. For example:

```
# look ok
my $sub = sub{
 ...
};

# not so ok
my @sub = sub{
 ...
},
$otherthing;
```

**Comment by sten22** [comment]
I think Is okay, several examples from cpan:

- https://monosnap.com/file/6exfcelhFpmNe5gsaJg73H5pKMv5Ox
- https://monosnap.com/file/hc36wNzDQhG8sFwnTEt0h3parnRhBx

```
sub {
    my @nestedMapGrep = map {

    }
    grep {

    }
    grep { };

    my @subs = sub {

    },
    sub {

    },
    1, 2, 3;

    my @rememberLevel = join ", ",
                        map { }
                        grep { },
                        @args;
}
```

**Comment by hurricup** [comment]
I guess that some examples from CPAN is not good argument. I'd prefer to rely on [Perl::Tidy](http://search.cpan.org/~shancock/Perl-Tidy/) or [perlstyle](https://perldoc.perl.org/perlstyle.html) or some other code style if there is any.

**Comment by hurricup** [comment]
Fixed with #1525 
Feel free to build from master and test
